src/Resources/TaxIdentificationNumber.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace MyParcelCom\ApiSdk\Resources;
+
+use MyParcelCom\ApiSdk\Exceptions\MyParcelComException;
+use MyParcelCom\ApiSdk\Resources\Traits\JsonSerializable;
+
+class TaxIdentificationNumber
+{
+    use JsonSerializable;
+
+    const EORI = 'eori';
+    const IOSS = 'ioss';
+    const VAT = 'vat';
+
+    /** @var string */
+    private $countryCode;
+
+    /** @var string */
+    private $number;
+
+    /** @var string|null */
+    private $description;
+
+    /** @var string */
+    private $type;
+
+    /**
+     * @param string $countryCode
+     * @return $this
+     */
+    public function setCountryCode($countryCode)
+    {
+        $this->countryCode = $countryCode;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCountryCode()
+    {
+        return $this->countryCode;
+    }
+
+    /**
+     * @param string $number
+     * @return $this
+     */
+    public function setNumber($number)
+    {
+        $this->number = $number;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getNumber()
+    {
+        return $this->number;
+    }
+
+    /**
+     * @param string|null $description
+     * @return $this
+     */
+    public function setDescription($description)
+    {
+        $this->description = $description;
+
+        return $this;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getDescription()
+    {
+        return $this->description;
+    }
+
+    /**
+     * @param string $type
+     * @return $this
+     */
+    public function setType($type)
+    {
+        if (!in_array($type, [self::EORI, self::IOSS, self::VAT])) {
+            throw new MyParcelComException('Invalid TaxIdentificationNumber type: ' . $type);
```

Comment:
Here we can then change the error message into something like `'Invalid TaxIdentificationNumber type: ' . $type . ' must be one of ' . join(', ', TaxIdentificationNumberTypeEnum::getValues())`.